packages/purpose-process/src/services/purposeService.ts

```diff
@@ -734,6 +734,340 @@ export function purposeServiceBuilder(
       return newPurposeVersion;
     },
 
+    async activatePurposeVersion({
+      purposeId,
+      versionId,
+      organizationId,
+      correlationId,
+      logger,
+    }: {
+      purposeId: PurposeId;
+      versionId: PurposeVersionId;
+      organizationId: TenantId;
+      correlationId: string;
+      logger: Logger;
+    }): Promise<PurposeVersion> {
+      logger.info(`Activating Version ${versionId} in Purpose ${purposeId}`);
+
+      const purpose = await retrievePurpose(purposeId, readModelService);
+      const purposeVersion = retrievePurposeVersion(versionId, purpose);
+
+      const eservice = await retrieveEService(
+        purpose.data.eserviceId,
+        readModelService
+      );
+
+      if (purposeVersion.state === purposeVersionState.draft) {
+        const riskAnalysisForm = purpose.data.riskAnalysisForm;
+
+        if (!riskAnalysisForm) {
+          throw missingRiskAnalysis(purposeId);
+        }
+
+        const tenant = await getInvolvedTenantByEServiceMode(
+          eservice,
+          purpose.data.consumerId,
+          readModelService
+        );
+
+        assertTenantKindExists(tenant);
+
+        validateRiskAnalysisOrThrow({
+          riskAnalysisForm:
+            riskAnalysisFormToRiskAnalysisFormToValidate(riskAnalysisForm),
+          schemaOnlyValidation: false,
+          tenantKind: tenant.kind,
+        });
+      }
+
+      const ownership = getOrganizationRole({
+        organizationId,
+        producerId: eservice.producerId,
+        consumerId: purpose.data.consumerId,
+      });
+
+      function changeToWaitForApprovalFromDraft(): {
+        event: CreateEvent<PurposeEvent>;
+        updatedPurposeVersion: PurposeVersion;
+      } {
+        const updatedPurposeVersion: PurposeVersion = {
+          ...purposeVersion,
```

Comment:
When you define inner functions and use variables that are not taken from the parameters, it's difficult to follow the logic. For example here `purposeVersion` is taken from the outer function and it's a bit difficult to understand what value this variable holds at this point of the flow.

Reply:
Yeah, I agree. I changed a bit the implementation moving those function outside the service